portal-api/src/v1/controllers/utilisation-report-service/due-reports.controller.js

```diff
@@ -0,0 +1,77 @@
+const { subMonths, isSameMonth, addMonths, format } = require('date-fns');
+const api = require('../../api');
+
+const getMostRecentReport = (reports) => {
+  if (!reports || reports.length === 0) {
+    return null;
+  }
+
+  let mostRecentYear = 0;
+  let mostRecentMonth = 0;
+  let mostRecentReportIndex = 0;
+  reports.forEach((report, index) => {
+    const { month, year } = report;
+    if (year > mostRecentYear) {
+      mostRecentYear = year;
+      mostRecentMonth = month;
+      mostRecentReportIndex = index;
+    } else if (month > mostRecentMonth) {
+      mostRecentMonth = month;
+      mostRecentReportIndex = index;
+    }
+  });
+  return reports[mostRecentReportIndex];
+};
+
+const getDueReportDates = (mostRecentReport) => {
+  const currentDate = new Date();
+  const currentReportDate = subMonths(currentDate, 1);
+
+  // If most recent report is empty, assume no reports (and therefore current report is due)
+  let dueReportDate;
```

Comment:
`dueReportDate` isn't a const though because I keep moving it forward in the below logic. I will refactor some of the below code because I agree with your previous comment but I think I still need `dueReportDate` to be a `let` so that I can modify it

Reply:
It seems that js is happy for me to modify a `const` date with `addMonths`...